plugin/helpers_single_node.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package plugin
+
+import "fmt"
+
+// RunOnSingleNode implements Helper.RunOnSingleNode
+func (p *HelpersImpl) RunOnSingleNode(id string, f func()) (bool, error) {
+	err := p.ensureServerVersion("5.12.0")
+	if err != nil {
+		return false, err
+	}
+
+	id = fmt.Sprintf("%s%s", RUN_SINGLE_NODE_KEY_PREFIX, id)
+	updated, appErr := p.API.KVCompareAndSet(id, nil, []byte("lock"))
+	if appErr != nil {
+		return false, appErr
+	}
+	if !updated {
+		return false, nil
+	}
+
+	f()
+
+	appErr = p.API.KVDelete(id)
```

Comment:
@hanzei Is a potential use case for this helper to run a recurring job on one of the plugin instances? If so, how do we handle the case where the plugin crashes during the long-running function, and the `id` key does not get deleted from the kvstore?

Reply:
I would like to deffer this question to @lieut-data. He has a better vision on HA helpers methods for plugins then I have.